### DISCUSSION TITLE
Options to enable/disable at compile time?

### DISCUSSION THREAD

**Discussion by aznlir88** [pekwm/pekwm#139] [Q&A]
I was reading through pewkwm faq(https://github.com/pekwm/pekwm/blob/master/doc/faq.md) and found such sentence about harbour funcionality:

"If a dockapp doesn't go into the harbour even you have it enabled at compile time(...)"

As I am years long Fluxbox user already and tend to disable funcionality I do not need as I compile it sounds very interesting. Personally I totally support being able not to compile in things I am not gonna use.

Can you please provide me with a list of features possible to enable/disable while compiling pekwm with information how to enable/disable them?

As I am new to pekwm and see it is once again in development I hope it will give us possibility to adjust it to our needs by choosing a lot of small features we can compile in or not. For example Fluxbox does not allow users not to compile in window tabbing that I do not use at all - hopefully pekwm will allow me not to compile in window grouping.

Very eager to hear back about it as I am trying pekwm as soon as I have some free time :).

**Comment by pekdon** [comment]
The list of options available is in part of the documentation, have a look at:

https://github.com/pekwm/pekwm/blob/master/doc/overview.md#compiling-pekwm

And no, there's no support for excluding grouping as part of compilation. That logic is spread throughout the codebase and would be difficult and error prone to exclude. There are things that could be excluded, such as autoproperties but that currently is not an option to exclude.

What's your main motivation to remove all these things, are your systems running low on resources? (20 year old hardware I have still pekwm fine)

**Reply by aznlir88** [reply]
Interesting as on that web page I see no word about disabling dock apps harbour - guess it should be added there.

About the reason, I just like having possibility not to waste resources on something I do not use. Using 10 year old PC myself but it is not about the need but about the possibility.

**Reply by pekdon** [reply]
IMHO possibility without a need is just wasted complexity as everything comes with a cost, be it complexity, man hours or risk of mis configuration.

So, pekwm has gotten fever options with things that can be disabled to reduce code complexity. Most options are based on external dependencies today.
